### DISCUSSION TITLE
Trying to add changelog support to a Python project

### DISCUSSION THREAD

**Discussion by strayer** [renovatebot/renovate#15769] [Request Help] [answered]
Hey,

I'm trying to add changelog support to a Python project. I added the relevant code to add `Source` and `Changelog` `project_urls` to the packaging and pushed test versions to TestPyPI: https://test.pypi.org/project/chia-blockchain-test/

But for some reason, this did not work. I tested this here: https://github.com/strayer/chia-project-urls-test/pull/1

As you can see, there is no changelog referenced. The linked repository has a changelog AND a Github release for this version:

https://github.com/Chia-Network/chia-blockchain/releases/tag/1.3.5
https://github.com/Chia-Network/chia-blockchain/blob/main/CHANGELOG.md#135-chia-blockchain-2022-5-11

Am I not doing this correctly? Based on the Renovate code I assumed this would work:
https://github.com/renovatebot/renovate/blob/098d22d09b099667bd96c2594a79098d0fdbe01a/lib/modules/datasource/pypi/index.ts#L114-L144

**Comment by rarkins** [comment, marked as answer]
You've configured it to use the simple registry, which doesn't include repository metadata

**Reply by strayer** [reply]
You mean the line `--extra-index-url https://test.pypi.org/simple/` in the requirements.txt? As far as I understand that is how to tell pip to use a different package index. I searched a bit and couldn't find anything on a different endpoint that has more metadata…

Renovate seems to default to `https://pypi.org/pypi/` and then use `<package_name>/json` to fetch metadata based on the file I linked in the original post. Changing the requirements.txt to `--extra-index-url https://test.pypi.org/pypi/` makes Renovate find the changelog (see https://github.com/strayer/chia-project-urls-test/pull/1 for reference).

Could this be a bug in Renovate? Pip needs the simple index to work:

```
Looking in indexes: https://pypi.org/simple, https://test.pypi.org/pypi/
ERROR: Could not find a version that satisfies the requirement chia-blockchain-test==0.0.1 (from versions: 1.3.5)
ERROR: No matching distribution found for chia-blockchain-test==0.0.1
```

**Reply by rarkins** [reply]
Not a bug

**Reply by strayer** [reply]
Thanks for taking time to look into this!

**Reply by rarkins** [reply]
In summary: simple registries do not contain metadata. If someone things we should always attempt to fall back to `../pypi` - or just for `pypi.org`, then it could be considered as a feature request.

In your case above you could also configure `registryUrls` manually to point to the `../pypi/` endpoint.

**Reply by viceice** [reply]
i think for pypi.org we should automatically use the pypi endpoint and ignore user defined simple endpoint completely

**Reply by strayer** [reply]
@rarkins thanks for explaining! To be fair, what I'm doing is pretty unusual, as I'm specifically pointing Renovate to a TestPyPI package to test the metadata resolved by Renovate. I'm unsure if this applies to other package indexes as well (as in: I have no idea if the ../pypi endpoint is "standard" or specific to pypi.org).
